### PR TITLE
Add non-record streaming legal TTT late-block submission

### PR DESCRIPTION
**Pull request by simon-marcus**
## Summary
This adds a non-record `track_10min_16mb` submission documenting a full 8xH100 run built from the March 23 leader with a different legal TTT protocol.

## Why submit this
This is intentionally framed as a non-record submission. The main value is to provide the community with a clean comparison between:
- the March 23 leader's chunked score-first legal TTT over all blocks
- a streaming legal TTT variant that adapts online and only updates the last 4 blocks

## What changed relative to the March 23 leader
- kept the same base model stack and `LeakyReLU(0.5)^2` activation
- switched eval-time adaptation to `TTT_MODE=stream`
- used `TTT_PARAM_MODE=late_blocks` with `TTT_LAST_N_BLOCKS=4`
- included explicit preflight and run logs for reproducibility

## Result
Final `legal_ttt_exact` on the included run:
- `1.12082320`

This slightly trails the March 23 leader's seed-1337 `1.11922988`, but the intermediate pre-TTT and sliding-window metrics are very close, which makes this a useful data point on the TTT protocol itself.

## Included artifacts
- `train_gpt.py`
- `train_seed1337.log`
- `runpod_launch.log`
- `preflight.log`
- `submission.json`

### CODE REVIEW

**Comment by MatoTeziTanka** [comment]
### Community Review — Add non-record streaming legal TTT late-block submission

**Compliance:** LOOKS CLEAN — legal score-first-per-chunk TTT (PR #1413 pattern)


---

## Analysis

### 1. N-gram family bug check (CLOSE trigger: target token in hash key)

`BigramHashEmbedding.bigram_hash` (line 705–711):

```python
out[..., 0] = mod
out[..., 1:] = torch.bitwise_xor(36313 * t[..., 1:], 27191 * t[..., :-1]) % mod
```

Position `i` (for i >= 1) hashes `tokens[i]` XOR `tokens[i-1]`. This means the embedding at position `i` depends on the **current** token `t[i]` — which is the input token, not the target. The target at position `i` is `tokens[i+1]`. No target leakage. Position 0 gets a fixed fallback (`mod`). This is the legal `BigramHash` pattern. **No CLOSE trigger.**

### 2. Pre-Quant TTT check (CLOSE trigger: multi-epoch AdamW on val_tokens without score-first)

TTT optimizer is `torch.optim.SGD` with momentum (line 1285), not AdamW. AdamW appears only in the training optimizers (lines 1762, 1778), which are unrelated to TTT eval. **AdamW CLOSE condition not met.**

### 3. Legal TTT check

Both modes implement score-first per chunk/window:

- **chunk mode** (line 1300–1373): `torch.inference_mode()` block scores all windows in a chunk first, accumulates metrics, then trains on that chunk. Scoring and training are cleanly separated. Score-first-per-chunk satisfied.

**Verdict:** LOOKS CLEAN — legal TTT implementation.

**Recommendation to @cocohearts @valerio-oai @0hq @yuzhougu-oai @notapplica:** **MERGE** pending the usual record-track checks.

---
*Reviewed by [@MatoTeziTanka](https://github.com/MatoTeziTanka) — [The Agora](https://matotezitanka.github.io/parameter-golf). Compliance audit via LLM agent (Sonnet) reviewing full train_gpt.py source. If this review misread your code, please call it out so I can re-audit manually.*
